extractor/filesystem/os/apk/extractor.go

```diff
@@ -129,67 +126,91 @@ func (e Extractor) Extract(ctx context.Context, input *filesystem.ScanInput) ([]
 	return inventory, err
 }
 
+// parseSingleApkRecord reads from the scanner a single record,
+// returns nil, nil when scanner ends.
+func parseSingleApkRecord(scanner *bufio.Scanner) (map[string]string, error) {
+	// There is currently 26 keys defined here (Under "Installed Database V2"):
+	// https://wiki.alpinelinux.org/wiki/Apk_spec
+	var group map[string]string = make(map[string]string, 30)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if line != "" {
+			key, val, found := strings.Cut(line, ":")
+
+			if !found {
+				return nil, fmt.Errorf("invalid line: %q", line)
+			}
+
+			group[key] = val
+			continue
+		}
+
+		// check both that line is empty and we have filled out data in group
+		// this avoids double empty lines returning early
+		if line == "" && len(group) > 0 {
+			return group, nil
+		}
+	}
+
+	// Err() could only be non nil when Scan() returns false
+	return group, scanner.Err()
+}
+
 func (e Extractor) extractFromInput(ctx context.Context, input *filesystem.ScanInput) ([]*extractor.Inventory, error) {
 	m, err := osrelease.GetOSRelease(input.FS)
 	if err != nil {
 		log.Errorf("osrelease.ParseOsRelease(): %v", err)
 	}
 
-	rd := textproto.NewReader(bufio.NewReader(input.Reader))
-	pkgs := []*extractor.Inventory{}
+	scanner := bufio.NewScanner(input.Reader)
+	inventories := make([]*extractor.Inventory, 0)
```

Comment:
I see the use of make in various places with small numbers. Is there any reason for doing so? I think this barely helps performance wise, but is harder to read as without make.
```
pkgs := []*extractor.Inventory{}
vs
inventories := make([]*extractor.Inventory, 0)
```

```
group := map[string]string{}
vs
var group map[string]string = make(map[string]string, 30)
```

I think in most of Scalibr code base we only use make when the number of elements is known and/or big.

Reply:
Not for any particular reason, I'm not really sure which version to use. I changed to not using make now.